lib/tx_build/signer.ex

```diff
@@ -44,7 +52,51 @@ defmodule Stellar.TxBuild.Signer do
     end
   end
 
-  @spec validate_signer_key(signer_key :: String.t()) :: validation()
+  @spec validate_signer_key_tuple(signer_key :: tuple()) :: validation()
+  defp validate_signer_key_tuple({:pre_auth_tx, signer_key}) do
+    with {:ok, raw_pre_auth_tx} <- Base.decode16(signer_key, case: :lower),
+         32 <- byte_size(raw_pre_auth_tx),
+         pre_auth_tx <- StrKey.encode!(raw_pre_auth_tx, :pre_auth_tx) do
```

Comment:
@miguelnietoa This logic can be encapsulated in a single function and then reused in the other use cases.

Reply:
As a side note, It took me some time to understand the code :P, IMO we can refactor it a little bit!

Reply:
Yes, I can create a function and be reused for `pre_auth_tx` and `hash_x` as they are hex-encoded values of 32 bytes!